### PR TITLE
Dynamically builds docker build matrix

### PR DESCRIPTION
**Pull request by adriancole**
Similar to https://github.com/golangci/golangci-lint/issues/828

### CODE REVIEW

**Comment by adriancole** [comment]
yay works now!

**Comment by adriancole** [comment]
cool verified this picked up ratpack w/o manual config change

File: .github/workflows/docker-tests.yml [hunk 1]
```diff
@@ -8,12 +8,25 @@ on:
...
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - id: set-matrix
+        name: Generate outputs.matrix from pom.xml listing
+        run: echo "::set-output name=matrix::{\"include\":$(.github/workflows/projects-as-json)}"
```
Comment (anuraaga) [inline .github/workflows/docker-tests.yml:23]: Pretty slick in a magical way

Comment (adriancole) [reply .github/workflows/docker-tests.yml:23]: 🧙

**Pull request merged**